### PR TITLE
rename PESQ

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

rename PESQ for consistency
part of #729 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: torchmetrics/audio/pesq.py [hunk 1]
```diff
@@ -100,7 +102,7 @@ def __init__(
         )
         if not _PESQ_AVAILABLE:
             raise ModuleNotFoundError(
-                "PESQ metric requires that pesq is installed."
+                "PerceptualEvaluationSpeechQuality metric requires that pesq is installed."
```
Comment (rohitgr7) [inline torchmetrics/audio/pesq.py:105]:
```suggestion
                "PerceptualEvaluationSpeechQuality metric requires that `pesq` is installed."
```

File: torchmetrics/audio/pesq.py [hunk 2]
```diff
@@ -120,11 +122,45 @@ def update(self, preds: Tensor, target: Tensor) -> None:  # type: ignore
...
     def compute(self) -> Tensor:
         """Computes average PESQ."""
         return self.sum_pesq / self.total
+
+
+class PESQ(PerceptualEvaluationSpeechQuality):
+    """Perceptual Evaluation of Speech Quality (PESQ).
+
+    .. deprecated:: v0.7
+        Use :class:`torchmetrics.audio.PermutationInvariantTraining`. Will be removed in v0.8.
```
Comment (rohitgr7) [inline torchmetrics/audio/pesq.py:141]:
```suggestion
        Use :class:`torchmetrics.audio.PerceptualEvaluationSpeechQuality`. Will be removed in v0.8.
```

**Review by rohitgr7 — APPROVED** [review]

File: torchmetrics/functional/audio/pesq.py [hunk 3]
```diff
@@ -98,3 +102,23 @@ def pesq(preds: Tensor, target: Tensor, fs: int, mode: str, keep_same_device: bo
         pesq_val = pesq_val.to(preds.device)
 
     return pesq_val
+
+
+@deprecated(target=perceptual_evaluation_speech_quality, deprecated_in="0.7", remove_in="0.8", stream=_future_warning)
+def pesq(preds: Tensor, target: Tensor, fs: int, mode: str, keep_same_device: bool = False) -> Tensor:
```
Comment (rohitgr7) [inline torchmetrics/functional/audio/pesq.py:108]: are there any deprecation tests?

Comment (Borda) [reply torchmetrics/functional/audio/pesq.py:108]: we are still in a pre-stable (v1.0) so we do not check the correct raised warning as I believe it works fine as we use it all other PL family packages :]

**Pull request merged**